youtube_dl/extractor/instagram.py

```diff
@@ -173,6 +188,27 @@ def _real_extract(self, url):
                 media = try_get(
                     additional_data, lambda x: x['graphql']['shortcode_media'],
                     dict)
+                if not media:
+                    # In some cases, the response from the server has items in it
+                    # Here is a sample of code that works fine with one item, but
+                    # I don't know for multiple items, cuz I don't have test links
+                    # for those.
+                    items0 = try_get(additional_data, lambda x: x['items'][0], dict)
+                    if items0:
+                        best_quality = items0.get("video_versions")[0]
+                        video_url = best_quality.get('url')
+                        height = int_or_none(best_quality.get('height'))
+                        width = int_or_none(best_quality.get('width'))
+                        description = items0.get('caption').get('text')
+                        title = None
```

Comment:
why None and not `title = self._og_search_title(webpage)` ?

Reply:
I'm clueless on the title so have passed `None`. By the way, while trying with your idea it leads to `ERROR: Unable to extract OpenGraph title; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.` So it is better to have `'Video by %s' % uploader_id` as the title or  `description`(caption) can be a title

Reply:
The test page has a perfectly good `application/ld+json` script element with `@type: VideoObject` which should be the default source for metadata.

Reply:
Have you tried the test page with yt-dlp? I'd rather pull changes from the downstream extractor than make new changes here, if that deals with the issue.

Reply:
Hey no I have no time to check is this happening with yt-dlp. But I have faced this problem in my application so I decide to share my opinion how I made it to work